scipy/optimize/_optimize.py

```diff
@@ -270,7 +270,20 @@ def __dir__(self):
 class OptimizeWarning(UserWarning):
     pass
 
-
+def _check_positive_definite(Hk):
+    def is_pos_def(A):
+        if np.allclose(A, A.T):
```

Comment:
You can use `linalg.issymmetric` instead here, to have a bit more performance. But it does an exact comparison of the off-diagonal elements so no tolerances. However if you provide tolerances it falls back to this test you have here.

Reply:
Thanks for the suggestion. I was not aware of this function. Will make the change for readability and consistency.